gossipsub.go

```diff
@@ -57,25 +81,39 @@ func NewGossipSub(ctx context.Context, h host.Host, opts ...Option) (*PubSub, er
 // is the fanout map. Fanout peer lists are expired if we don't publish any
 // messages to their topic for GossipSubFanoutTTL.
 type GossipSubRouter struct {
-	p       *PubSub
-	peers   map[peer.ID]protocol.ID         // peer protocols
-	mesh    map[string]map[peer.ID]struct{} // topic meshes
-	fanout  map[string]map[peer.ID]struct{} // topic fanout
-	lastpub map[string]int64                // last publish time for fanout topics
-	gossip  map[peer.ID][]*pb.ControlIHave  // pending gossip
-	control map[peer.ID]*pb.ControlMessage  // pending control messages
-	mcache  *MessageCache
-	tracer  *pubsubTracer
+	p            *PubSub
+	peers        map[peer.ID]protocol.ID                 // peer protocols
+	mesh         map[string]map[peer.ID]struct{}         // topic meshes
+	fanout       map[string]map[peer.ID]struct{}         // topic fanout
+	lastpub      map[string]int64                        // last publish time for fanout topics
+	gossip       map[peer.ID][]*pb.ControlIHave          // pending gossip
+	control      map[peer.ID]*pb.ControlMessage          // pending control messages
+	backoff      map[string]map[peer.ID]time.Time        // prune backoff
+	connect      chan connectInfo                        // px connection requests
+	connectMx    sync.Mutex                              // mutex protecting connectPend
+	connectPend  map[peer.ID]*routing.SignedRoutingState // pending px connections
+	connectSched chan struct{}                           // px connection trigger
+	mcache       *MessageCache
+	tracer       *pubsubTracer
+}
+
+type connectInfo struct {
+	p   peer.ID
+	srr *routing.SignedRoutingState
```

Comment:
This is really cryptic, I had to go back and look at the full patch to figure out what these were. Can we write them out?